canvas_modules/common-canvas/src/common-properties/controls/dropdown/dropdown.jsx

```diff
@@ -265,6 +280,13 @@ class DropDown extends React.Component {
 					disabled={this.props.state === STATES.DISABLED || this.disableEmptyListDropdown}
 					type="default"
 					items={dropDown.options}
+					itemToString={(item) => (item ? (
+						<div className="properties-dropdown-label">
+							<div>{ item.label }</div>
+							<div className="custom-icon">{this.getItemIcon()}</div>
```

Comment:
What if there is no icon returned?  I don't think we should add the div if the callback returns a null or undefined.